src/Hazelcast.Net/Serialization/Compact/CompactSerializationSerializer.cs

```diff
@@ -70,6 +71,13 @@ public CompactSerializationSerializer(CompactOptions options, ISchemas schemas,
 
         public bool HasRegistrationForType(Type type) => _registrationsByType.ContainsKey(type);
 
+        public bool TryGetSerializer(Type type, out ICompactSerializer? serializer)
+        {
+            var hasSerializer = _registrationsByType.TryGetValue(type, out var registration);
+            serializer = hasSerializer ? registration!.Serializer?.Serializer : null;
```

Comment:
I'm confused here. So, there could be a serializer but null?

Reply:
so... `_registrationsByType.TryGetValue(type, out var registration);`  is going to return `true` and set `registration` if there is a registration, otherwise return `false`. And then, the registration *may* have a serializer, if one has been specified, or none, if no serializer has been specified and we're going to use the reflection-based serializer. So:
```
serializer = hasSerializer  // as returned by the line above

  // we know registration is not null, but registration.Serializer may be null
  ? registration!.Serializer?.Serializer  

  // we don't have a registration, hence we don't have a serializer
  : null; 
```

That being said, the code being tricky, I am going to verify again that this is correct. Will update here.